web_m2x_options/static/src/js/form.js

```diff
@@ -120,161 +120,173 @@ odoo.define('web_m2x_options.web_m2x_options', function (require) {
 
         _search: function (search_val) {
             var self = this;
-            var def = $.Deferred();
-            this.orderer.add(def);
-
-            // add options limit used to change number of selections record
-            // returned.
-            if (!_.isUndefined(this.ir_options['web_m2x_options.limit'])) {
-                this.limit = parseInt(this.ir_options['web_m2x_options.limit'], 10);
-            }
-
-            if (typeof this.nodeOptions.limit === 'number') {
-                this.limit = this.nodeOptions.limit;
-            }
+            var def = new Promise(function (resolve) {
+                // add options limit used to change number of selections record
+                // returned.
+                if (!_.isUndefined(self.ir_options['web_m2x_options.limit'])) {
+                    self.limit = parseInt(self.ir_options['web_m2x_options.limit'], 10);
```

Comment:
```suggestion
                    this.limit = parseInt(self.ir_options['web_m2x_options.limit'], 10);
```

The same for all other `self`s